### PR TITLE
stm32f0l0g0: fixes for SPI half duplex

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

- stm32f0l0g0/hardware/stm32_spi.h: remove unused definitions
- stm32f0l0g0/stm32_spi.c: add missing SPI mode config and fix ifdef
- stm32f0l0g0/stm32_spi.c: remove unused spi_readbyte function
- stm32f0l0g0/stm32_spi.c: fix receiving data for half duplex mode

## Impact
rx for SPI half duplex works correctly now

## Testing
read data from MEMS sensor in SPI 3-wire mode

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/src/stm32f0l0g0/hardware/stm32_spi.h [hunk 1]
```diff
@@ -96,42 +96,6 @@
...
-#  define STM32_SPI6_CR2          (STM32_SPI6_BASE + STM32_SPI_CR2_OFFSET)
-#  define STM32_SPI6_SR           (STM32_SPI6_BASE + STM32_SPI_SR_OFFSET)
-#  define STM32_SPI6_DR           (STM32_SPI6_BASE + STM32_SPI_DR_OFFSET)
-#  define STM32_SPI6_CRCPR        (STM32_SPI6_BASE + STM32_SPI_CRCPR_OFFSET)
-#  define STM32_SPI6_RXCRCR       (STM32_SPI6_BASE + STM32_SPI_RXCRCR_OFFSET)
-#  define STM32_SPI6_TXCRCR       (STM32_SPI6_BASE + STM32_SPI_TXCRCR_OFFSET)
-#  define STM32_SPI6_I2SCFGR      (STM32_SPI6_BASE + STM32_SPI_I2SCFGR_OFFSET)
-#  define STM32_SPI6_I2SPR        (STM32_SPI6_BASE + STM32_SPI_I2SPR_OFFSET)
-#endif
-
```
Comment (acassis) [inline arch/arm/src/stm32f0l0g0/hardware/stm32_spi.h:134]: @raiden00pl are you sure none STM32L0xx, STM32F0xx, STM32L0xx, STM32G0xx have SPI4/5/6 ?

Comment (raiden00pl) [reply arch/arm/src/stm32f0l0g0/hardware/stm32_spi.h:134]: I see now that ST has introduced STM32G0Bx and STM32G0Cx chips that have SPI3. I'll restore SPI3 definitions and add support for it in the next PR.

File: arch/arm/src/stm32f0l0g0/stm32_spi.c [hunk 2]
```diff
@@ -428,6 +433,62 @@ static inline void spi_putreg(struct stm32_spidev_s *priv,
...
+ *
+ ****************************************************************************/
+
+static inline void spi_rx_mode(struct stm32_spidev_s *priv, bool enable)
+{
+  if (enable == true)
+    {
+      /* Enable RX */
+
+      if (priv->rx_mode == false)
```
Comment (pkarashchenko) [inline arch/arm/src/stm32f0l0g0/stm32_spi.c:450]:
```suggestion
      if (!priv->rx_mode)
```

File: arch/arm/src/stm32f0l0g0/stm32_spi.c [hunk 3]
```diff
@@ -428,6 +433,62 @@ static inline void spi_putreg(struct stm32_spidev_s *priv,
...
+ * Name: spi_rx_mode
+ *
+ * Description:
+ *   Activate SPI RX or SPI TX for the half-duplex mode
+ *
+ ****************************************************************************/
+
+static inline void spi_rx_mode(struct stm32_spidev_s *priv, bool enable)
+{
+  if (enable == true)
```
Comment (pkarashchenko) [inline arch/arm/src/stm32f0l0g0/stm32_spi.c:446]:
```suggestion
  if (enable)
```

File: arch/arm/src/stm32f0l0g0/stm32_spi.c [hunk 4]
```diff
@@ -428,6 +433,62 @@ static inline void spi_putreg(struct stm32_spidev_s *priv,
...
+          spi_modifycr(STM32_SPI_CR1_OFFSET, priv, SPI_CR1_SPE, 0);
+
+          priv->rx_mode = true;
+        }
+    }
+  else
+    {
+      /* Enable TX */
+
+      if (priv->rx_mode == true)
```
Comment (pkarashchenko) [inline arch/arm/src/stm32f0l0g0/stm32_spi.c:473]:
```suggestion
      if (priv->rx_mode)
```

File: arch/arm/src/stm32f0l0g0/stm32_spi.c [hunk 5]
```diff
@@ -1326,12 +1356,32 @@ static uint32_t spi_send(struct spi_dev_s *dev, uint32_t wd)
...
+      spi_writeword(priv, (uint16_t)(wd & 0xffff));
+      ret = (uint32_t)spi_readword(priv);
+    }
+  else
+    {
+      /* In half duplex we must send data and receive data in separate
+       * spi_send() calls.
+       */
+
+      if (priv->rx_now == false)
```
Comment (pkarashchenko) [inline arch/arm/src/stm32f0l0g0/stm32_spi.c:1374]:
```suggestion
      if (!priv->rx_now)
```

Comment (raiden00pl) [reply arch/arm/src/stm32f0l0g0/stm32_spi.c]: done

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**